// @stackline/angular-multiselect-dropdown/src/app/angular2-multiselect-dropdown/clickOutside.js
System.register(["@angular/core"], function (exports_1, context_1) {
    "use strict";
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var core_1, ClickOutsideDirective, ScrollDirective, styleDirective;
    var __moduleName = context_1 && context_1.id;
    return {
        setters: [
            function (core_1_1) {
                core_1 = core_1_1;
            }
        ],
        execute: function () {
            ClickOutsideDirective = /** @class */ (function () {
                function ClickOutsideDirective(_elementRef) {
                    this._elementRef = _elementRef;
                    this.clickOutside = new core_1.EventEmitter();
                }
                ClickOutsideDirective.prototype.onClick = function (event, targetElement) {
                    if (!targetElement) {
                        return;
                    }
                    var clickedInside = this._elementRef.nativeElement.contains(targetElement);
                    if (!clickedInside) {
                        this.clickOutside.emit(event);
                    }
                };
                __decorate([
                    core_1.Output(),
                    __metadata("design:type", Object)
                ], ClickOutsideDirective.prototype, "clickOutside", void 0);
                __decorate([
                    core_1.HostListener('document:click', ['$event', '$event.target']),
                    core_1.HostListener('document:touchstart', ['$event', '$event.target']),
                    __metadata("design:type", Function),
                    __metadata("design:paramtypes", [MouseEvent, HTMLElement]),
                    __metadata("design:returntype", void 0)
                ], ClickOutsideDirective.prototype, "onClick", null);
                ClickOutsideDirective = __decorate([
                    core_1.Directive({
                        selector: '[clickOutside]'
                    }),
                    __metadata("design:paramtypes", [core_1.ElementRef])
                ], ClickOutsideDirective);
                return ClickOutsideDirective;
            }());
            exports_1("ClickOutsideDirective", ClickOutsideDirective);
            ScrollDirective = /** @class */ (function () {
                function ScrollDirective(_elementRef) {
                    this._elementRef = _elementRef;
                    this.scroll = new core_1.EventEmitter();
                }
                ScrollDirective.prototype.onClick = function (event, targetElement) {
                    this.scroll.emit(event);
                };
                __decorate([
                    core_1.Output(),
                    __metadata("design:type", Object)
                ], ScrollDirective.prototype, "scroll", void 0);
                __decorate([
                    core_1.HostListener('scroll', ['$event']),
                    __metadata("design:type", Function),
                    __metadata("design:paramtypes", [MouseEvent, HTMLElement]),
                    __metadata("design:returntype", void 0)
                ], ScrollDirective.prototype, "onClick", null);
                ScrollDirective = __decorate([
                    core_1.Directive({
                        selector: '[scroll]'
                    }),
                    __metadata("design:paramtypes", [core_1.ElementRef])
                ], ScrollDirective);
                return ScrollDirective;
            }());
            exports_1("ScrollDirective", ScrollDirective);
            styleDirective = /** @class */ (function () {
                function styleDirective(el) {
                    this.el = el;
                }
                styleDirective.prototype.ngOnInit = function () {
                    this.el.nativeElement.style.top = this.styleVal;
                };
                styleDirective.prototype.ngOnChanges = function () {
                    this.el.nativeElement.style.top = this.styleVal;
                };
                __decorate([
                    core_1.Input('styleProp'),
                    __metadata("design:type", Number)
                ], styleDirective.prototype, "styleVal", void 0);
                styleDirective = __decorate([
                    core_1.Directive({
                        selector: '[styleProp]'
                    }),
                    __metadata("design:paramtypes", [core_1.ElementRef])
                ], styleDirective);
                return styleDirective;
            }());
            exports_1("styleDirective", styleDirective);
        }
    };
});
