// @stackline/angular-multiselect-dropdown/src/app/angular2-multiselect-dropdown/menu-item.js
System.register(["@angular/core"], function (exports_1, context_1) {
    "use strict";
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var __metadata = (this && this.__metadata) || function (k, v) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
    };
    var __moduleName = context_1 && context_1.id;
    var core_1, Item, Badge, TemplateRenderer;
    return {
        setters: [
            function (core_1_1) {
                core_1 = core_1_1;
            }
        ],
        execute: function () {
            Item = /** @class */ (function () {
                function Item() {
                }
                __decorate([
                    core_1.ContentChild(core_1.TemplateRef),
                    __metadata("design:type", core_1.TemplateRef)
                ], Item.prototype, "template", void 0);
                Item = __decorate([
                    core_1.Component({
                        selector: 'c-item',
                        template: ""
                    }),
                    __metadata("design:paramtypes", [])
                ], Item);
                return Item;
            }());
            exports_1("Item", Item);
            Badge = /** @class */ (function () {
                function Badge() {
                }
                __decorate([
                    core_1.ContentChild(core_1.TemplateRef),
                    __metadata("design:type", core_1.TemplateRef)
                ], Badge.prototype, "template", void 0);
                Badge = __decorate([
                    core_1.Component({
                        selector: 'c-badge',
                        template: ""
                    }),
                    __metadata("design:paramtypes", [])
                ], Badge);
                return Badge;
            }());
            exports_1("Badge", Badge);
            TemplateRenderer = /** @class */ (function () {
                function TemplateRenderer(viewContainer) {
                    this.viewContainer = viewContainer;
                }
                TemplateRenderer.prototype.ngOnInit = function () {
                    this.view = this.viewContainer.createEmbeddedView(this.data.template, {
                        '\$implicit': this.data,
                        'item': this.item
                    });
                };
                TemplateRenderer.prototype.ngOnDestroy = function () {
                    this.view.destroy();
                };
                __decorate([
                    core_1.Input(),
                    __metadata("design:type", Object)
                ], TemplateRenderer.prototype, "data", void 0);
                __decorate([
                    core_1.Input(),
                    __metadata("design:type", Object)
                ], TemplateRenderer.prototype, "item", void 0);
                TemplateRenderer = __decorate([
                    core_1.Component({
                        selector: 'c-templateRenderer',
                        template: ""
                    }),
                    __metadata("design:paramtypes", [core_1.ViewContainerRef])
                ], TemplateRenderer);
                return TemplateRenderer;
            }());
            exports_1("TemplateRenderer", TemplateRenderer);
        }
    };
});
