// @stackline/angular-multiselect-dropdown/src/app/angular2-multiselect-dropdown/list-filter.js
System.register(["@angular/core"], function (exports_1, context_1) {
    "use strict";
    var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    };
    var core_1, ListFilterPipe;
    var __moduleName = context_1 && context_1.id;
    return {
        setters: [
            function (core_1_1) {
                core_1 = core_1_1;
            }
        ],
        execute: function () {
            ListFilterPipe = /** @class */ (function () {
                function ListFilterPipe() {
                }
                ListFilterPipe.prototype.transform = function (items, filter) {
                    var _this = this;
                    if (!items || !filter) {
                        return items;
                    }
                    return items.filter(function (item) { return _this.applyFilter(item, filter); });
                };
                ListFilterPipe.prototype.applyFilter = function (item, filter) {
                    var found = false;
                    for (var prop in item) {
                        if (filter) {
                            if (item[prop].toString().toLowerCase().indexOf(filter.toLowerCase()) >= 0) {
                                found = true;
                            }
                        }
                        else
                            found = true;
                    }
                    return found;
                };
                ListFilterPipe = __decorate([
                    core_1.Pipe({
                        name: 'listFilter',
                        pure: false
                    })
                ], ListFilterPipe);
                return ListFilterPipe;
            }());
            exports_1("ListFilterPipe", ListFilterPipe);
        }
    };
});
